smartmodules/jolt/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+use once_cell::sync::OnceCell;
+
+use eyre::ContextCompat;
+use fluvio_jolt::TransformSpec;
+use fluvio_smartmodule::{
+    dataplane::smartmodule::SmartModuleExtraParams, smartmodule, Record, RecordData, Result,
+};
+
+static SPEC: OnceCell<TransformSpec> = OnceCell::new();
+
+#[smartmodule(init)]
+fn init(params: SmartModuleExtraParams) -> Result<()> {
+    if let Some(raw_spec) = params.get("spec") {
+        match serde_json::from_str(raw_spec) {
+            Ok(spec) => {
+                SPEC.set(spec).expect("spec is already initialized");
+                Ok(())
+            }
+            Err(err) => {
+                eprintln!("unable to parse spec from params: {:?}", err);
+                Err(eyre::Report::msg(
```

Comment:
This should return missing param error.  See regex example

Reply:
Done